src/GCL/Deserializer.cpp

```diff
@@ -77,6 +83,16 @@ GCL::Deserializer::Deserializer( const std::filesystem::path& rPath )
 		_read( FileDescriptor, m_pFileBuffer, static_cast< uint32_t >( m_FileSize ) );
 		_close( FileDescriptor );
 	}
+	#else
+	int FileDescriptor = open(rPath.c_str(), O_RDONLY);
+	m_FileSize = static_cast< size_t >( lseek( FileDescriptor, 0, SEEK_END ) );
+	lseek( FileDescriptor, 0, SEEK_SET );
+
+	m_pFileBuffer = static_cast< char* >( malloc( m_FileSize ) );
+
+	read( FileDescriptor, m_pFileBuffer, static_cast< uint32_t >( m_FileSize ) );
+	close( FileDescriptor );
+	#endif
```

Comment:
Here we can probably add some defines on windows to make sure we don't duplicate the code. Something like:
```cpp
#define open _wsopen
#define lseek _lseek
// etc..
```

Reply:
Shouldn't be a problem for most, though some windows equivalents have different signatures